### PR TITLE
Buttonify

### PR DESCRIPTION
**Pull request by yoshuawuyts**
This implements part of https://github.com/datproject/dat-desktop/issues/280,
namely buttons and icons. It creates a slightly more intuitive API to create
elements, and decouples the icons from the buttons itself making them more
flexible. All elements should be accessible and reusable between projects.

The app runs and all event seem to work as expected which means I caught all
instances I think. Thanks!

### API
```js
var button = require('./elements/button')
var icon = require('./elements/icon')

icon('exit')                    // render an exit icon
icon('exit', { class: 'pt3' })  // render an exit icon with extra classes

button('some-button')                                      // render a plain button
button.green('my-cool-button', { onclick: handleclick })   // render a green button with an onclick handler
button.icon('my-cooler-button', { icon: icon('exit') })    // render an accessible icon-only button
button('some-button', { class: 'pr2 mt3' })                // render a plain button with extra classes
```

### CODE REVIEW

**Comment by yoshuawuyts** [comment]
hah, found some beautiful glitches - hold on with the merge for a lil bit ok?

File: elements/dat-import.js [hunk 1]
```diff
@@ -64,19 +64,14 @@ function datImportElement (props) {
...
-
   return html`
     <label for="dat-import" class="relative dib pa0 b--none ${prefix}">
       <input name="dat-import"
         type="text"
         placeholder="Import dat"
         onkeydown=${onKeyDown}
         class="input-reset">
-      ${linkIcon}
+      ${icon('link', { class: 'absolute top-0 bottom-0 left-0' })}
```
Comment (juliangruber) [inline elements/dat-import.js]: what about this syntax?

```js
icon('#link.absolute.top-0.bottom-0.left-0')
```

Comment (yoshuawuyts) [inline elements/dat-import.js]: Oh I kinda like that; very close to [hyperscript's class / id shorthand](https://github.com/hyperhype/hyperscript#classes--id) - I think having that at some point would definitely make sense. Nice :sparkles:

I'd probably make sure there'd be full consistency with hyperscript tho:
```js
icon('link.absolute.top-0.bottom-0.left-0#some-id')
```
but yeah that's nit picking. I like it!

Comment (karissa) [inline elements/dat-import.js]: I like having the 'class' dict, it also is just as simple and already works.

Comment (yoshuawuyts) [inline elements/dat-import.js]: annnnd... package! - https://github.com/yoshuawuyts/parse-class

Comment (yoshuawuyts) [inline elements/dat-import.js]: Yeah nah, agree - this would only work for icons right now; buttons is harder - gonna leave this for a bit

**Comment by yoshuawuyts** [comment]
@Kriesse ran into an SVG styling issue; not sure what changed but could you perhaps take a look - styling SVGs has never been my forte. Once that's fixed this should be ready to merge.

screen shot 2017-03-02 at 10 19 23

**Comment by Kriesse** [comment]
@yoshuawuyts the intro screen and a few other small misalignments are fixed, ready to merge from my side! Preview:
image

**Comment by yoshuawuyts** [comment]
Awesome, just ran through it and it's looking :100: - merging!

**Pull request merged**